lib/iris/util.py

```diff
@@ -696,6 +696,27 @@ def _build_full_slice_given_keys(keys, ndim):
     return full_slice
 
 
+def _slice_data_with_keys(data, keys, n_parent_dims):
+    # Combine _build_full_slice_given_keys and column_slices_generator to index
```

Comment:
This new routine should probably have its own test.
I could write something now, or we could wait until we decide whether this becomes part of DataManager ?